### PR TITLE
Indexing a sample json

### PR DESCRIPTION
**Pull request by ShahinaKP**
Hi @nishintn 

Please review  and mereg the branch for the implementation:

- Indexing a sample json(Replica of SetInfo screen in Optimize)
   - Grouped the data with first letter 
   - Implement click action on the alphabets and display the matching content on left.

### CODE REVIEW

File: src/app/components/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
   templateUrl: './dashboard.component.html',
-  styleUrls: ['./dashboard.component.scss']
+  styleUrls: ['./dashboard.component.scss'],
+  providers: [ ReturnSetInfoDataService ]
 })
-export class DashboardComponent {  
-  constructor() {}  
-  
+export class DashboardComponent {
+  data: Observable<Array<any>>;
```
Comment (mad-nishin) [inline src/app/components/dashboard/dashboard.component.ts]: Observable<Array<any>>; What is this syntax?

File: src/app/components/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+
+  constructor(private dataService: ReturnSetInfoDataService) {
+    dataService.getSetInfoData().subscribe(res => {
+      this.data = res;
+      debugger
```
Comment (mad-nishin) [inline src/app/components/dashboard/dashboard.component.ts]: debugger!!!!!

File: src/app/components/dashboard/dashboard.component.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        if (!this.groupedData[firstLetter]) {
+          this.groupedData[firstLetter] = [];
+          this.index.push(firstLetter);
+
+        }
+        
+        this.groupedData[firstLetter].push(gropedItem);       
+  
+      });
+      console.log(this.index);
```
Comment (mad-nishin) [inline src/app/components/dashboard/dashboard.component.ts]: Get rid of Console stms.

File: src/app/components/dashboard/dashboard.component.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        
+        this.groupedData[firstLetter].push(gropedItem);       
+  
+      });
+      console.log(this.index);      
+    });
+        
+  }  
+  
+  indexTo(index) {
```
Comment (mad-nishin) [inline src/app/components/dashboard/dashboard.component.ts:39]: From where did you get this code?

File: src/app/components/dashboard/dashboard.service.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { Http, Response } from '@angular/http';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class ReturnSetInfoDataService {
+    constructor(private http: Http) {}
+
+    getSetInfoData() : Observable<any> {
+        return this.http.get('../../assets/json/setInfo.json')
```
Comment (mad-nishin) [inline src/app/components/dashboard/dashboard.service.ts]: the url should  moved to a const.

**Review by mad-nishin — COMMENTED** [review]
When you are done with changes, Please merge. Need to wait.

File: src/app/components/dashboard/dashboard.component.ts (hunk 1 above)
Comment (ShahinaKP) [reply src/app/components/dashboard/dashboard.component.ts]: @nishintn 
Haven't used this and already edited before.

File: src/app/components/dashboard/dashboard.component.ts (hunk 2 above)
Comment (ShahinaKP) [reply src/app/components/dashboard/dashboard.component.ts]: @nishintn 

Here also same. The debugger was removed and the code is not seen in my local.

File: src/app/components/dashboard/dashboard.component.ts (hunk 4 above)
Comment (ShahinaKP) [reply src/app/components/dashboard/dashboard.component.ts:39]: @nishintn 
Code for grouping with first letter was taken from a stackoverflow answer (https://stackoverflow.com/questions/34259272/linq-js-grouping-objects-with-there-name-first-letter).
Build other features with this.

File: src/app/components/dashboard/dashboard.service.ts (hunk 5 above)
Comment (ShahinaKP) [reply src/app/components/dashboard/dashboard.service.ts]: @nishintn 
Done

File: src/app/components/dashboard/dashboard.component.ts (hunk 3 above)
Comment (ShahinaKP) [reply src/app/components/dashboard/dashboard.component.ts]: @nishintn 
Removed.

**Pull request merged**